hazelcast/src/test/java/com/hazelcast/cache/CacheCreationTest.java

```diff
@@ -0,0 +1,115 @@
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.concurrent.CountDownLatch;
+
+
+@RunWith(HazelcastSerialClassRunner.class)
+@Category(SlowTest.class)
+public class CacheCreationTest {
+
+    private final URL configUrl1 = getClass().getClassLoader().getResource("test-hazelcast-real-jcache.xml");
+    Config hzConfig;
+
+    @Before
+    public void init() throws Exception{
+        Hazelcast.shutdownAll();
+        XmlConfigBuilder configBuilder = new XmlConfigBuilder(configUrl1.getFile());
+        hzConfig = configBuilder.build();
+    }
+
+    @After
+    public void cleanup() throws Exception{
+        Hazelcast.shutdownAll();
+    }
+
+    @Test
+    public void creatingASingleCache() throws URISyntaxException {
+
+        HazelcastInstance hazelcastInstance = Hazelcast.newHazelcastInstance(hzConfig);
+        HazelcastServerCachingProvider cachingProvider = HazelcastServerCachingProvider
+                .createCachingProvider(hazelcastInstance);
+
+        Cache<Object, Object> cache = cachingProvider.getCacheManager().getCache("xmlCache" + 1);
+        cache.get(1);
+    }
+
+
+    @Test
+    public void creatingASingleCacheFromMultiProviders() throws URISyntaxException {
+
+        int threadCount=4;
+
+        final CountDownLatch latch = new CountDownLatch(threadCount);
+        for (int i = 0; i < threadCount; i++) {
+            final int finalI = i;
+            new Thread() {
+                public void run() {
+                    HazelcastInstance hazelcastInstance = Hazelcast.newHazelcastInstance(hzConfig);
+                    HazelcastServerCachingProvider cachingProvider = HazelcastServerCachingProvider
+                            .createCachingProvider(hazelcastInstance);
+
+                    Cache<Object, Object> cache = cachingProvider.getCacheManager().getCache("xmlCache" );
+                    cache.get(1);
+                    latch.countDown();
+                }
+            }.start();
+        }
+        HazelcastTestSupport.assertOpenEventually(latch);
+    }
+
+    @Test
+    public void creatingMultiCacheFromMultiProviders() throws URISyntaxException {
+
+        int threadCount=4;
+
+        final CountDownLatch latch = new CountDownLatch(threadCount);
+        for (int i = 0; i < threadCount; i++) {
+            final int finalI = i;
+            new Thread() {
+                public void run() {
+                    HazelcastInstance hazelcastInstance = Hazelcast.newHazelcastInstance(hzConfig);
+                    HazelcastServerCachingProvider cachingProvider = HazelcastServerCachingProvider
+                            .createCachingProvider(hazelcastInstance);
+
+                    Cache<Object, Object> cache = cachingProvider.getCacheManager().getCache("xmlCache"+finalI);
+                    cache.get(1);
+                    latch.countDown();
+                }
+            }.start();
+        }
+        HazelcastTestSupport.assertOpenEventually(latch);
```

Comment:
@serkan-ozal Could you consult me how it could be done? Because I can't find suitable form for it in our code. 
I quite understand what exactly should be checked BUT your advise is highly appreciated.

p.s. the same as for previous comment.